hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/upgrade/BasicUpgradeFinalizer.java

```diff
@@ -436,4 +456,15 @@ private void logAndThrow(Exception e, String msg, ResultCodes resultCode)
   protected void updateLayoutVersionInDB(V vm, T comp) throws IOException {
     throw new UnsupportedOperationException();
   }
+
+  protected void postFinalizeUpgrade() throws IOException {
+  }
+
+  protected void finalizeVersionManager(Storage storageConfig)
```

Comment:
nit. Can the method name be just finalizeUpgrade (to go with pre and post)?

Reply:
Given that we are refactoring the abstractions here, I think we can also make org.apache.hadoop.ozone.upgrade.BasicUpgradeFinalizer#finalize abstract and mandate subclasses to implement it.

Reply:
done